pkg/virt-launcher/virtwrap/device/sriov/hostdev.go

```diff
@@ -193,3 +193,45 @@ func deviceLookup(hostDevices []api.HostDevice, deviceAlias string) *api.HostDev
 	}
 	return nil
 }
+
+type deviceAttacher interface {
+	AttachDeviceFlags(xmlData string, flags libvirt.DomainDeviceModifyFlags) error
+}
+
+func AttachHostDevices(dom deviceAttacher, hostDevices []api.HostDevice) error {
+	var errs []error
+	for _, hostDev := range hostDevices {
+		if err := attachHostDevice(dom, hostDev); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		return buildHostPlugHostDevicesErrorMessage(errs)
+	}
+
+	return nil
+}
+
+func attachHostDevice(dom deviceAttacher, hostDev api.HostDevice) error {
+	devXML, err := xml.Marshal(hostDev)
+	if err != nil {
+		return fmt.Errorf("failed to encode (xml) host-device %v, err: %v", hostDev, err)
+	}
+	err = dom.AttachDeviceFlags(string(devXML), libvirt.DOMAIN_DEVICE_MODIFY_LIVE | libvirt.DOMAIN_DEVICE_MODIFY_CONFIG)
+	if err != nil {
+		return fmt.Errorf("failed to attach host-device %s, err: %v", devXML, err)
```

Comment:
Printing the whole `devXML`  and not just the alias of the device is "by design" or a typo?

>> The error will include the Alias of each device that
>> failed to attach.
I'm asking since that's what was written on the commit message.

Reply:
It should print the device XML, its  small few lines.
I will change the commit message sorry